src/components/Counter.js

```diff
@@ -0,0 +1,48 @@
+import React, { useState, Fragment, useEffect } from 'react';
+
+import { db } from '../lib/firebase';
+
+import { useCollectionData } from 'react-firebase-hooks/firestore';
+
+const Counter = () => {
+  const [counter, setCounter] = useState(
+    JSON.parse(localStorage.getItem('counter')) || 1,
+  );
+
+  const [countersCollection] = useCollectionData(db.collection('counters'), {
+    idField: 'id',
+  });
+
+  useEffect(() => {
+    localStorage.setItem('counter', JSON.stringify(counter));
+  }, [counter]);
+
+  const sortedCollection =
+    countersCollection &&
+    [...countersCollection].sort((a, b) => a.value - b.value);
+
+  const handleClick = () => {
+    setCounter((c) => c + 1);
+    db.collection('counters').add({
+      value: counter,
+    });
+  };
+
+  return (
+    <Fragment>
```

Comment:
I really like the usage of <Fragment> here 😁  There is also a shorter syntax, in case you did not know (<>). But using Fragment is of course fine.